opencti-platform/opencti-graphql/src/manager/publisherManager.ts

```diff
@@ -31,6 +94,212 @@ const OUTCOME_TYPE_UI = 'UI';
 const OUTCOME_TYPE_EMAIL = 'EMAIL';
 const OUTCOME_TYPE_WEBHOOK = 'WEBHOOK';
 
+const extractStixRepresentative = async (context: AuthContext, user: NotificationUser, stix: StixObject): Promise<string> => {
+  const entityType = stix.extensions[STIX_EXT_OCTI].type;
+  // region Modules
+  const convertFn = getStixRepresentativeConverters(entityType);
+  if (convertFn) {
+    return convertFn(stix);
+  }
+  // endregion
+  // region Sighting
+  if (isStixSightingRelationship(entityType)) {
```

Comment:
Why did you move this method? It deals with stix, I think it makes sense to have it in the stix file

Reply:
I put it in an other file because of dependencies cycles error in imports.

Reply:
Indeed we had this cycle : middleware.js > engine.js > identifier.js > stix-converter.js > stixCoreObject.js > middleware.js (since I added the relation stix-converter > stixCoreObject by using the findById method in stix-converter)